client/src/main/java/com/paypal/selion/platform/grid/SeleniumGridListener.java

```diff
@@ -123,10 +123,11 @@ public void beforeInvocation(IInvokedMethod method, ITestResult testResult) {
                     System.exit(1);  // NOSONAR
                 }
             }
+        } catch (RuntimeException e) { // NOSONAR
+            Grid.getThreadLocalException().set(e);
         } catch (Exception e) { // NOSONAR
```

Comment:
This PR does not fix the issue when a checked exception occurs in `beforeInvocation`.. Since a message is only log when an `Exception` is caught,`@WebTest` and/or `@MobileTest` execution continue and a `NullPointerException` will happen in the first call into `Grid`.. Furthermore, the true failure (the original Exception), is not including with the test results.